subprojects/language-java/src/integTest/groovy/org/gradle/java/compile/incremental/AbstractSourceIncrementalCompilationIntegrationTest.groovy

```diff
@@ -50,395 +49,350 @@ class SourceIncrementalJavaCompilationIntegrationTest extends AbstractIntegratio
 
         then:
         outputs.recompiledClasses 'A', 'A$InnerA'
     }
 
     def "detects transitive dependencies"() {
-        java "class A {}", "class B extends A {}", "class C extends B {}", "class D {}"
-        outputs.snapshot { run "compileJava" }
+        source "class A {}", "class B extends A {}", "class C extends B {}", "class D {}"
+        outputs.snapshot { run language.compileTaskName }
 
         when:
-        java "class A { /* change */ }"
-        run "compileJava"
+        source "class A { /* change */ }"
+        run language.compileTaskName
 
         then:
         outputs.recompiledClasses 'A', 'B', 'C'
 
         when:
         outputs.snapshot()
-        java "class B { /* change */ }"
-        run "compileJava"
+        source "class B { /* change */ }"
+        run language.compileTaskName
 
         then:
         outputs.recompiledClasses 'B', 'C'
     }
 
     def "detects transitive dependencies with inner classes"() {
-        java "class A {}", "class B extends A {}", "class D {}"
-        java """class C extends B {
+        source "class A {}", "class B extends A {}", "class D {}"
+        source """class C extends B {
             class InnerC {}
         }
         """
-        outputs.snapshot { run "compileJava" }
+        outputs.snapshot { run language.compileTaskName }
 
         when:
-        java "class A { /* change */ }"
-        run "compileJava"
+        source "class A { /* change */ }"
+        run language.compileTaskName
 
         then:
         outputs.recompiledClasses 'A', 'B', 'C', 'C$InnerC'
     }
 
     def "handles cycles in class dependencies"() {
-        java "class A {}", "class D {}"
-        java "class B extends A { C c; }", "class C extends B {}" //cycle
-        outputs.snapshot { run "compileJava" }
+        source "class A {}", "class D {}"
+        source "class B extends A { C c; }", "class C extends B {}" //cycle
+        outputs.snapshot { run language.compileTaskName }
 
         when:
-        java "class A { /* change */ }"
-        run "compileJava"
+        source "class A { /* change */ }"
+        run language.compileTaskName
 
         then:
         outputs.recompiledClasses 'A', 'B', 'C'
     }
 
-    @Unroll
-    def "change to #retention retention annotation class recompiles #desc"() {
-        def annotationClass = file("src/main/java/SomeAnnotation.java") << """
-            import java.lang.annotation.*;
-
-            @Retention(RetentionPolicy.$retention) 
-            public @interface SomeAnnotation {}
-        """
-        java "@SomeAnnotation class A {}", "class B {}"
-        outputs.snapshot { run "compileJava" }
-
-        when:
-        annotationClass.text += "/* change */"
-        run "compileJava"
-
-        then:
-        outputs.recompiledClasses(expected as String[])
-
-        where:
-        desc              | retention | expected
-        'all'             | 'SOURCE'  | ['A', 'B', 'SomeAnnotation']
```

Comment:
This scenaio fails on Groovy compilation because Groovy compiler surprisingly doesn't keep `@Retention` in class files:

![image](https://user-images.githubusercontent.com/12689835/59660982-6a136780-91dc-11e9-9203-a757bf211e5c.png)